docs/contribute/general/git.rst

```diff
@@ -49,7 +49,26 @@ So this file must at least include the following lines:
 ***************
 
 Before merging a PR all commits shall be squashed into few (desired only
-one) logical commits.
+one) logical commits. This is done following the rules below.
+
+Use ``Squash & Merge`` in case:
+
+   #. there is one initial commit + n "fix findings" commits
+
+      **AND**
+   #. all commits by the same author.
+
+Use ``Rebase & Merge`` or ``Merge Commit`` in case:
+
+   #. the commits address different topics
+      (sort by topics and combine each topic to a single commit)
+
+      **OR**
+   #. the commits have different authors
```

Comment:
Our own old discussion is slightly wrong here. We often have PRs with commits by different authors that we squash. That happens rather frequently when collaborating or even pair programming. I know, crazy. But sometimes we collaborate 🗡️ 

VS-Code and GitHub-UI will typically auto-append `Co-authored-by: name <name@example.com>` in such cases.

Reply:
I learnt today that it is demanded by one committer company that it has to be clearly stated which parts belong to which company for IP topics, liability and so.

Reply:
I'm sorry, but that needs to be clarified further as we typically work in pair programming across companies.

Reply:
And I see your updated PR. But it doesn't quite solve the problem 100%. As the Author vs Co-Author are somewhat random for each commit.